openmetadata-service/src/main/java/org/openmetadata/service/resources/topics/TopicResource.java

```diff
@@ -347,6 +349,32 @@ public Topic addSampleData(
     return addHref(uriInfo, topic);
   }
 
+  @GET
+  @Path("/{id}/sampleData")
```

Comment:
added the GET sample data endpoint to follow the same approach as with tables. This helps us validate the permissions for masking sample data